### PR TITLE
#12 Shift Basics Batch Execution

### PR DESCRIPTION
**Pull request by albertochimentiinbibo**
Linked to #12 

This PR adds the Shift Basics section for Batch execution to the shift documentation

### CODE REVIEW

File: docs/getting_started/basics/batch.md
Comment (hectorjimenezinbibo) [inline docs/getting_started/basics/batch.md:15]:
```suggestion
The execution of this method will return the `SWorkflow` Python class containing the workflow and a Python dictionary containing the information on the Python modules loaded by the workflow. Have a look at the [Shift API](../../reference/api.md) for more information on the method and the `SWorkflow` class.
```

File: docs/getting_started/basics/batch.md
Comment (hectorjimenezinbibo) [inline docs/getting_started/basics/batch.md:3]: Just a question, I have seen different ways when we mention Python, in some cases (different pages) is written as `Python` and in other cases as `python`, it's a little bit inconsistent.
Can we decide which is the best way to keep consistency between the different pages? @Inbibo/shift

**Review by hectorjimenezinbibo — CHANGES_REQUESTED** [review]

File: docs/getting_started/basics/batch.md
Comment (juliavisserinbibo) [inline docs/getting_started/basics/batch.md:25]: I would add an example of this in the code

File: docs/getting_started/basics/batch.md
Comment (JorgeHernandezibanezinbibo) [inline docs/getting_started/basics/batch.md:15]: @juliavisserinbibo should consider the catalog_manifest returned like a dict of "Python modules loaded" by Shift, or a dict of "Shift Catalogs"? I know that a catalog is a pyModule at the end, but just to be sure how to refer to the catalog in the documentation.

File: docs/getting_started/basics/batch.md
Comment (JorgeHernandezibanezinbibo) [inline docs/getting_started/basics/batch.md:25]: I would suggest to add another example with parameters. Maybe:
```suggestion
External inputs for the workflow can be set by providing a Python dictionary as first argument of the `workflow.execute` call. This dictionary should be formatted using the input plug names from the `Input` operator as keys and the content that should be passed as values.

```python
workflowExternals = {
    "myBooleanPlug": True,
    "myStringPlug": "Hello world"
}
result = workflow.execute(workflowExternals)
```
```

File: docs/getting_started/basics/batch.md
Comment (JorgeHernandezibanezinbibo) [inline docs/getting_started/basics/batch.md:27]: I would add and explanation of how to check if the workflow execution went well. Explaining that if the result is empty, probably means that the workflow execution have fail (If you set the stopOnError=True)

**Review by JorgeHernandezibanezinbibo — CHANGES_REQUESTED** [review]
Just some suggestions mainly about the execution part

File: docs/getting_started/basics/batch.md
Comment (juliavisserinbibo) [reply docs/getting_started/basics/batch.md:15]: Right, it should be
```suggestion
The execution of this method will return the `SWorkflow` Python class containing the workflow and a Python dictionary containing the information on the Shift catalogs loaded by the workflow. Have a look at the [Shift API](../../reference/api.md) for more information on the method and the `SWorkflow` class.
```

File: docs/getting_started/basics/batch.md
Comment (albertochimentiinbibo) [reply docs/getting_started/basics/batch.md:25]: done

File: docs/getting_started/basics/batch.md
Comment (albertochimentiinbibo) [reply docs/getting_started/basics/batch.md:15]: right! done!

File: docs/getting_started/basics/batch.md
Comment (albertochimentiinbibo) [reply docs/getting_started/basics/batch.md:25]: done!

File: docs/getting_started/basics/batch.md
Comment (albertochimentiinbibo) [reply docs/getting_started/basics/batch.md:27]: Added a subsection to explain the stop on error behaviour and explain the result content.
done!

File: docs/getting_started/basics/batch.md
Comment (juliavisserinbibo) [inline docs/getting_started/basics/batch.md:46]: Avoid using  the word "this" when possible, especially at the start of new blocks of text.
```suggestion
By setting the `stopOnError` argument to `True` the workflow will stop executing when an error is raised. In case the execution stops the `result` will return `None`.
```

File: docs/getting_started/basics/batch.md
Comment (juliavisserinbibo) [inline docs/getting_started/basics/batch.md:50]:
```suggestion
It is highly recommended to close the workflow after the execution is done. This process will take care of cleaning the unneeded Shift catalogs from memory. This can be performed by calling the method:
```

File: docs/getting_started/basics/batch.md
Comment (juliavisserinbibo) [inline docs/getting_started/basics/batch.md:15]: Avoid using "this"

```suggestion
The `openBatchWorkflow `method will return the `SWorkflow` Python class containing the workflow and a Python dictionary containing the information on the Shft catalogs loaded by the workflow. Have a look at the [Shift API](../../reference/api.md) for more information on the method and the `SWorkflow` class.
```

File: docs/getting_started/basics/batch.md
Comment (juliavisserinbibo) [inline docs/getting_started/basics/batch.md:38]: I would be more specific, otherwise it might sound like the general execution of the workflow can be changed

```suggestion
### Stop on Error Behaviour
```

**Review by juliavisserinbibo — CHANGES_REQUESTED** [review]

File: docs/getting_started/basics/batch.md
Comment (albertochimentiinbibo) [reply docs/getting_started/basics/batch.md:46]: Shouldn't we just remove this part now that @JorgeHernandezibanezinbibo changed the default value?

File: docs/getting_started/basics/batch.md
Comment (albertochimentiinbibo) [reply docs/getting_started/basics/batch.md:38]: I vote for removing this section now that we changed the default value. No need to specify the behaviour anymore I think

File: docs/getting_started/basics/batch.md
Comment (JorgeHernandezibanezinbibo) [reply docs/getting_started/basics/batch.md:46]: I'm waiting for the approval of the PR, but if we agree on merging it. I think this can be skip yes, because now is not a thing required to be done always.

File: docs/getting_started/basics/batch.md
Comment (JorgeHernandezibanezinbibo) [reply docs/getting_started/basics/batch.md]: I think you have a typo, pls check to fix "Shft catalogs" by Shift catalogs @albertochimentiinbibo

Comment (albertochimentiinbibo) [reply docs/getting_started/basics/batch.md]: great catch man thanks a lot!

File: docs/getting_started/basics/batch.md
Comment (albertochimentiinbibo) [reply docs/getting_started/basics/batch.md]: removed it

File: docs/getting_started/basics/batch.md
Comment (albertochimentiinbibo) [reply docs/getting_started/basics/batch.md]: removed it

**Review by hectorjimenezinbibo — APPROVED** [review]

File: docs/getting_started/basics/batch.md
Comment (juliavisserinbibo) [inline docs/getting_started/basics/batch.md:1]: Should we add a TODO somewhere to remember to add the docs for the shift_batch command for the terminal? Once it is clear that it can work correctly.

Comment (albertochimentiinbibo) [reply docs/getting_started/basics/batch.md:1]: I added the shift_batch execution part in this commit https://github.com/Inbibo/shift_docs/pull/34/commits/10a7e798b8c09f7050fa84651a50ecfa257c7794

Please have a look if you can

File: docs/getting_started/basics/batch.md
Comment (juliavisserinbibo) [inline docs/getting_started/basics/batch.md:17]: We said that the same happens in Linux right? At the end we are using Bash

Comment (albertochimentiinbibo) [reply docs/getting_started/basics/batch.md:17]: yep, just tested on WSL and it also removes the quotation marks from the inputs.

**Review by juliavisserinbibo — APPROVED** [review]

**Review by mromeoinbibo — APPROVED** [review]

**Pull request merged**
